### PR TITLE
matrix-conduit: 0.4.0 -> 0.5.0

### PR DESCRIPTION
**Pull request by yu-re-ka**
###### Description of changes

Involves some hacks

https://github.com/NixOS/nixpkgs/issues/202762
https://github.com/rust-lang/cargo/issues/11192
https://github.com/ruma/ruma/issues/1441

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/matrix-conduit/default.nix [hunk 1]
```diff
@@ -2,16 +2,25 @@
...
+    sha256 = "sha256-GSCpmn6XRbmnfH31R9c6QW3/pez9KHPjI99dR+ln0P4=";
   };
 
-  cargoSha256 = "sha256-vE44I8lQ5VAfZB4WKLRv/xudoZJaFJGTT/UuumTePBU=";
+  # https://github.com/rust-lang/cargo/issues/11192
+  # https://github.com/ruma/ruma/issues/1441
+  depsExtraArgs.postBuild = ''
+    sed 's/serde = { workspace = true/serde = { features = [ "derive" ], version = "*"/g' -i $name/ruma*/Cargo.toml
+    sed 's/tracing = { workspace = true.*/tracing = { version = "*" }/g' -i $name/ruma*/Cargo.toml
+    sed 's/workspace = true/version = "*"/g' -i $name/ruma*/Cargo.toml
```
Comment (peterhoeg) [inline pkgs/servers/matrix-conduit/default.nix:19]: Minor nitpick, but I'd do this which I think is easier to read:

``` sh
    sed -i $name/ruma*/Cargo.toml \
      -e 's/serde = { workspace = true/serde = { features = [ "derive" ], version = "*"/g' \
      -e 's/tracing = { workspace = true.*/tracing = { version = "*" }/g' \
      -e 's/workspace = true/version = "*"/g'
```

Comment (peterhoeg) [reply pkgs/servers/matrix-conduit/default.nix:19]: That is of course assuming we cannot simply remove "workspace = true".

**Comment by peterhoeg** [comment]
One more thing - we should also change the module as the service can now run with `Type = "notify";`

File: pkgs/servers/matrix-conduit/default.nix [hunk 2]
```diff
@@ -1,17 +1,29 @@
...
 
-  cargoSha256 = "sha256-vE44I8lQ5VAfZB4WKLRv/xudoZJaFJGTT/UuumTePBU=";
+  # https://github.com/rust-lang/cargo/issues/11192
+  # https://github.com/ruma/ruma/issues/1441
+  postPatch = ''
+    pushd $cargoDepsCopy
+    sed -i ruma*/Cargo.toml \
+      -e 's/serde = { workspace = true/serde = { features = [ "derive" ], version = "*"/g' \
+      -e 's/tracing = { workspace = true.*/tracing = { version = "*" }/g' \
+      -e 's/workspace = true/version = "*"/g'
```
Comment (SuperSandro2000) [inline pkgs/servers/matrix-conduit/default.nix:21]: I would rather use a patch here to make the change more obvious and fail if some part does not apply with the next update.

Comment (peterhoeg) [reply pkgs/servers/matrix-conduit/default.nix:21]: That's a very good point. I'm with @SuperSandro2000 on this.

**Comment by yu-re-ka** [comment]
I don't know how to do this, since it is not part of the source code of the application. Either way, someone else can finish this.

**Comment by piegamesde** [comment]
Continuation in #212036

**Pull request closed without merging**